integration/test/performancetest/performance_test.go

```diff
@@ -1,59 +1,158 @@
-// go:build linux && integration
+//go:build linux && integration
 // +build linux,integration
 
 package performancetest
 
-import(
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"sync"
 	"testing"
 	"time"
-	"log"
-	"context"
+
 	"github.com/aws/amazon-cloudwatch-agent/integration/test"
 )
 
 const (
-	configPath = "resources/config.json"
 	configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
-	agentRuntimeMinutes = 5 //20min desired but 5mins for testing purposes 
+	agentRuntimeMinutes = 5 //20 mins desired but 5 mins for testing purposes
 	DynamoDBDataBase = "CWAPerformanceMetrics"
-
 )
 
 func TestPerformance(t *testing.T) {
+	configFiles := []string {
+		"./resources/config10Logs.json",
+		"./resources/config100Logs.json",
+		"./resources/config1000Logs.json"
+	}
+
+	tpsVals := []int {
+		10,
+		100,
+		1000
+	}
+	
 	agentContext := context.TODO()
 	instanceId := test.GetInstanceId()
 	log.Printf("Instance ID used for performance metrics : %s\n", instanceId)
 
-	test.CopyFile(configPath, configOutputPath)
+	//data base
+	dynamoDB := InitializeTransmitterAPI(DynamoDBDataBase) //add cwa version here
+	if dynamoDB == nil {
+		t.Fatalf("Error: generating dynamo table")
+	}
+
+	//run tests
+	for _, configPath := range configFiles {
+		for _, tps := range tpsVals {
+			t.Run(fmt.Sprintf("config file location: %s", configPath), func(t *testing.T) {
+				test.CopyFile(configPath, configOutputPath)
+	
+				test.StartAgent(configOutputPath, true)
+	
+				agentRunDuration := agentRuntimeMinutes * time.Minute
+	
+				StartLogWrite(t, agentRunDuration, configPath, tps)
+	
+				log.Printf("Agent has been running for : %s\n", (agentRunDuration).String())
+				test.StopAgent()
+	
+				//collect data
+				data, err := GetPerformanceMetrics(instanceId, agentRuntimeMinutes, agentContext, configPath)
+				//@TODO check if metrics are zero remove them and make sure there are non-zero metrics existing
+				if err != nil {
+					log.Println("Error: ", err)
+					t.Fatalf("Error: %v", err)
+				}
+	
+				if data == nil {
+					t.Fatalf("No data")
+				}
+				
+				_, err = dynamoDB.SendItem(data)
```

Comment:
this is all data that goes to the same hash/sort key, isn't it? How are we persisting this data so that we can differentiate between the different tests?

Reply:
Same question for me 👍